wurst/objects/abilities/TrollUpgrade.wurst

```diff
@@ -95,6 +96,11 @@ function upgradeUnit(unit origin, int targetID)
     let agi = origin.getAgi(false)
     let inT = origin.getInt(false)
 
+    // Once the unit transform into a subclass, medallion of courage is researched
+    // this way the super sub spell gets unlocked
+    if UNSUB_TROLL_UNIT_LIST.has(origin.getTypeId()) and GetPlayerTechResearched(origin.getOwner(), UPGD_MEDALLION_COURAGE, true) == false
```

Comment:
I don't really see the point of this check. Whenever a troll sub-classes, it should always have the medallion upgrade. Having a check would make sense if there were more than three tiers of classes, but I doubt we need to worry about that.

Also, checking for the research is pointless. Just research it anyway (unless that causes a bug or something). So at the very least you only need the check on the `UNSUB_TROLL_UNIT_LIST`.